requests/structures.py

```diff
@@ -103,3 +104,94 @@ def __getitem__(self, key):
 
     def get(self, key, default=None):
         return self.__dict__.get(key, default)
+
+
+class TimedCache(object):
```

Comment:
So, this is basically a `collections.MutableMapping`! We should make it *actually* be one, because that gives us a number of handy-dandy convenience methods.